### ISSUE TITLE
Intellisense for bower.json and package.json doesn't work properly

### ISSUE THREAD

**Issue by nhathongly** [aspnet/Tooling#219]
I installed VS2015 with ASP.Net 5 (beta 7) few days ago to learn ASP.Net 5. When I try to add dependencies in bower.json or package.json, intellisense doesn't work for listing out matching packages. It does work for the version in package.json. In bower.json, it shows "*", something like:

"dependencies": {
          "bootstrap": "*"
  }

I found the same question here
http://stackoverflow.com/questions/33065860/visual-studio-2015-no-bower-packages-in-intellisense

Sorry if I got anything wrong :P

**Comment by BillHiebert** [comment]
@madskristensen

**Comment by sayedihashimi** [comment]
TFS: 168977
I've added this to our internal TFS, we will reply back here soon.

**Comment by sanseref** [comment]
My intellisense isn't working on my new asp.net web app project either in VS15.

**Comment by palves78** [comment]
Try this in VS 2015 with your Solution/Project Open

Menu PROJECT->Refresh Project Toolbox Items

Did it work?
Cheers

**Comment by palves78** [comment]
For me this happens when my solution is under Git, if I copy my solution folder to another place and then open it, Intellisense starts to work again in both project.json and bower.json files.

**Comment by ben-wallis** [comment]
I'm also having issues with dependencies intellisense in both package.json and project.json in VS2015 Update 2. After adding a new line and beginning to type the name of the dependency within around half a second VS will lock up entirely and freeze for 2-3 minutes, after which what I've typed will appear really quickly (as if some time-out has expired) at which point intellisense still doesn't function for the dependency name but does for the version.

**Comment by ben-wallis** [comment]
Since I'm well aware of how unhelpful a bug report of "it freezes" with no additional info is, I did a bit of digging and it seems to be a network time-out related to the fetching of intellisense results.

I fired up WireShark and pinpointed the Amazon IP address (52.23.180.84 in the below screenshots) being used by the intellisense engine for NPM packages, and reproduced the bug twice. Both times resulted in an exact pause of 100 seconds (which directly corresponded to VS going to Not Responding) before a RST TCP response is received to the SSL request sent to the intellisense service, at which point VS starts working again.

![WireShark1](http://i.imgur.com/ZhR8IQ2.png)
![WireShark2](http://i.imgur.com/D8gbojG.png)

**Comment by palves78** [comment]
Hi, in the editor just use the correct schema for the right file type, for instance if you are editing the project.json file make sure the correct schema is being used.

**Comment by ToddGrun** [comment]
Regardless of what schema is selected, we shouldn't be freezing. @mlorbetske , was this fixed?

**Comment by mlorbetske** [comment]
@ToddGrun yep, this TFS bug for this was resolved for the upcoming RC2 release.
